cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs

```diff
@@ -296,10 +317,39 @@ txFeatureMismatch :: CardanoEra era
 txFeatureMismatch era feature =
     left (ShelleyTxCmdTxFeatureMismatch (anyCardanoEra era) feature)
 
-validateTxIns :: CardanoEra era
-              -> [TxIn]
-              -> ExceptT ShelleyTxCmdError IO [TxIn]
-validateTxIns _ = return -- no validation or era-checking needed
+validateTxIns
+  :: forall era. IsCardanoEra era
+  => CardanoEra era
+  -> [(TxIn, Maybe ScriptFile)]
+  -> ExceptT ShelleyTxCmdError IO [(TxIn, BuildTxWith BuildTx (Witness WitCtxTxIn era))]
+validateTxIns era = mapM convert
+ where
+   convert
+     :: (TxIn, Maybe ScriptFile)
+     -> ExceptT ShelleyTxCmdError IO (TxIn, BuildTxWith BuildTx (Witness WitCtxTxIn era))
+   convert (txin, mScriptFile) =
+     case mScriptFile of
+       Just (ScriptFile sFp) ->
+         case auxScriptsSupportedInEra era of
```

Comment:
So nothing to do with aux scripts. Otherwise the right idea.

One way you can see that from the types here is that you never use the `Just _supported` evidence. That's because the evidence is unnecessary! :smile:  The way we designed the API means that you'd need the evidence when you need to use the feature.